go/cmd/dolt/dolt.go

```diff
@@ -728,7 +728,15 @@ If you're interested in running this command against a remote host, hit us up on
 
 	if noValidRepository && isValidRepositoryRequired {
 		return func(ctx context.Context) (cli.Queryist, *sql.Context, func(), error) {
-			return nil, nil, nil, fmt.Errorf("The current directory is not a valid dolt repository.")
+			err := fmt.Errorf("The current directory is not a valid dolt repository.")
+			if rootEnv.DBLoadError == nbs.ErrUnsupportedTableFileFormat {
```

Comment:
Nit: Prefer `errors.Is(rootEnv.DBLoadError, nbs.ErrUnsupportedTableFileFormat)`, I think.